File: 12-05-2024_Largest_Local_Values_in_a_Matrix.py
class Solution:
    def largestLocal(self, grid: List[List[int]]) -> List[List[int]]:
        n = len(grid)
        maxLocal = [[0] * (n - 2) for _ in range(n - 2)]
        
        for i in range(n - 2):
            for j in range(n - 2):
                largestValue = 0
                for k in range(i, i + 3):
                    for z in range(j, j + 3):
                        largestValue = max(largestValue, grid[k][z])
                maxLocal[i][j] = largestValue
        
        return maxLocal

File: 12-05-2024_Largest_Local_Values_in_a_Matrix.cpp
class Solution {
public:
    vector<vector<int>> largestLocal(vector<vector<int>>& grid) {
        int n= grid.size();
        vector<vector<int>>maxLocal(n-2,vector<int>(n-2,0));
        int largest_Value=0;
        for(int i=0;i<n-2;i++){
            for(int j=0;j<n-2;j++){
                  for(int k=i;k<=i+2;k++){
                    for(int z=j;z<=j+2;z++){
                        largest_Value= max(largest_Value,grid[k][z]);
                    }
                  }
            maxLocal[i][j]=largest_Value;
            largest_Value=0;
            }
        }
        return maxLocal;
    }
};
